pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -203,15 +202,13 @@ func (r *ReconcileZookeeperCluster) reconcileStatefulSet(instance *zookeeperv1be
 				For details see:
 				https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically
 			*/
-			r.skipSTSReconcile++
-			if r.skipSTSReconcile < 6 {
-				r.log.Info("Waiting for Config Map update to sync...Skipping STS Reconcile")
-				return nil
+			for i := 0; i < 4; i++ {
```

Comment:
Why did you choose 2 minutes? Can it be less? In any case, let's not conflate issues. Issue #103 is about removing the global state (through skipSTSReconcile), which this PR does. it is separate question, but not less important, whether we can drop the wait time or even avoid it altogether.

This mechanism for notifying a scale-down is not very elegant, and we should look into replacing it with one that is more reliable and does not induce long wait times.

Reply:
`skipSTSReconcile` was added so that we could wait for the volume to be updated ....so this is directly related to scale down.
2 mins is the max sync time expected by kubernetes for volume updates based on this:
https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically

I agree that this is not a 100% reliable way to handle scale down, but it should work almost always and all the other approaches I have tried did not work due to kubernetes operational dynamics (see PR description). The worst case down side of this fix is that the deleted node would continue to stay in the ensemble and would have to be manually deleted by the admin.

Storing CLUSTER_SIZE in a zookeeper node feels like a little very complicated solution for handling a simple scale down and also introduces additional external dependency. It would require storing one value per cluster handled by the zk-operator.

Reply:
> skipSTSReconcile was added so that we could wait for the volume to be updated ....so this is directly related to scale down.

Ok, but let's agree that the main reason for having this PR is to avoid a global variable like `skipSTSReconcile`, which causes problems in the presence of multiple zk clusters. If my understanding is incorrect, then I 

> I agree that this is not a 100% reliable way to handle scale down

Can you enumerate the low-probability cases in which it fails? As part of answering this, let's step and review and what's happening:

1- In the case a zk cluster is being scaled-down, there is a zookeeper reconfig command executed against the zk ensemble. That's the first thing that happens.
2- The second step is to reconfigure the stateful set to have the correct number of replicas (pods).

If 1 and 2 are correct, including the order, then the zk ensemble will in principle work fine after 1, but step 2 might take time to execute, which is fine because it is not safety issue, it is just taking time to execute.

I'm also wondering what happens in the case of multiple consecutive scale down events (e.g., two severs are removed separately and consecutively).